core/src/com/fallenflame/game/PauseMode.java

```diff
@@ -107,8 +113,13 @@ public boolean touchDragged(int screenX, int screenY, int pointer) {
 
     @Override
     public boolean mouseMoved(int screenX, int screenY) {
-        for (int i = 0, j = hoverRects.length; i < j; i++) {
-            hoverStates[i] = (hoverRects[i].contains(screenX, screenY)) ? 1 : 0;
+        try {
+            for (int i = 0, j = hoverRects.length; i < j; i++) {
+            if (hoverRects[i] == null) continue;
+                hoverStates[i] = (hoverRects[i].contains(screenX, screenY)) ? 1 : 0;
+            }
+        } catch (NullPointerException ignored) {
+            Arrays.fill(hoverStates, 0);
```

Comment:
This is where it failed before, so this may be all we need. But what is the behavior here if `Arrays.fill(hoverStates, 0);` is executed? How does this impact the pause screen?

Reply:
This basically reset all the buttons so that they are rendered as normal or "not hovered".

Reply:
It's cause mouseMoved is an event-based state and the issue isn't hoverStates, but rather hoverRects, as that's an array of rectangles and is NOT filled before render. My guess is what's happening is if you move the mouse before render() completes one iteration, it crashes. Since mouseMoved is event-based, mouseMoved will take precedent over render, which is not what we want.

Reply:
I think this fix will actually fix the bug since I don't think we want to do an Arrays.fill() on hoverRects given what it's meant to do. The other option is to put an Arrays.fill() on hoverRects in the initialization of pauseMode